crypto/cmp/cmp_http.c

```diff
@@ -55,24 +55,45 @@ OSSL_CMP_MSG *OSSL_CMP_MSG_http_perform(OSSL_CMP_CTX *ctx,
 
     if (ctx->serverPort != 0)
         BIO_snprintf(server_port, sizeof(server_port), "%d", ctx->serverPort);
-
     tls_used = OSSL_CMP_CTX_get_http_cb_arg(ctx) != NULL;
-    ossl_cmp_log2(DEBUG, ctx, "connecting to CMP server %s%s",
-                  ctx->server, tls_used ? " using TLS" : "");
-    rsp = OSSL_HTTP_transfer(NULL, ctx->server, server_port,
+    if (ctx->http_ctx == NULL)
+        ossl_cmp_log3(DEBUG, ctx, "connecting to CMP server %s:%s%s",
+                      ctx->server, server_port, tls_used ? " using TLS" : "");
+
+    type = req->body->type;
+    keep_alive = ctx->keep_alive;
+    if (keep_alive != 0
```

Comment:
Suggestion : extract this `if` into a separate method. like 
```
/* really enable keep alive depending on request type */
keep_alive = should_keep_alive(ctx->keep_alive, req->body->type);
```
you will spare the `type` local variable

Reply:
Good thought :+1:
Done